Updated to 0.19.0 and issues starting Node-Red

Hi, after updating the

Here is what Node-Red is showing:

 Feb 22:50:14 - [info] Started flows


1 Feb 22:50:14 - [info] [zigbee2mqtt-server:zigbee2mqtt Server] MQTT Connected


1 Feb 22:50:14 - [info] [zigbee2mqtt-server:zigbee2mqtt Server] Refreshing devices


1 Feb 22:50:14 - [info] [mqtt-broker:MQTT Server] Connected to broker: mqtt://10.0.0.2:1883


1 Feb 22:50:14 - [red] Uncaught Exception:


1 Feb 22:50:14 - TypeError: Cannot read property 'split' of undefined


    at MqttSubscription.callback (/data/node_modules/node-red-contrib-zigbee2mqtt-devices/dist/nodes/hue-dimmer/devices-hue.js:13:49)


    at MqttSubscription.invokeIfMatch (/data/node_modules/node-red-contrib-zigbee2mqtt-devices/dist/lib/mqtt.js:28:18)


    at subscriptions.(anonymous function).forEach.sub (/data/node_modules/node-red-contrib-zigbee2mqtt-devices/dist/zigbee2mqtt-config.js:33:25)


    at Array.forEach (<anonymous>)


    at BridgeConfigConstructor.broker.subscribe (/data/node_modules/node-red-contrib-zigbee2mqtt-devices/dist/zigbee2mqtt-config.js:32:36)


    at MqttClient.handler (/usr/src/node-red/node_modules/@node-red/nodes/core/network/10-mqtt.js:320:25)


    at MqttClient.emit (events.js:198:13)


    at MqttClient._handlePublish (/usr/src/node-red/node_modules/mqtt/lib/client.js:1277:12)


    at MqttClient._handlePacket (/usr/src/node-red/node_modules/mqtt/lib/client.js:410:12)


    at work (/usr/src/node-red/node_modules/mqtt/lib/client.js:321:12)


    at process._tickCallback (internal/process/next_tick.js:61:11)


npm ERR! code ELIFECYCLE


npm ERR! errno 1


npm ERR! node-red-docker@1.2.7 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`


npm ERR! Exit status 1




 


npm ERR! Failed at the node-red-docker@1.2.7 start script.


npm ERR! This is probably not a problem with npm. There is likely additional logging output above.




npm ERR! A complete log of this run can be found in:


npm ERR!     /data/.npm/_logs/2021-02-01T11_50_14_083Z-debug.log




> node-red-docker@1.2.7 start /usr/src/node-red


> node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"




1 Feb 22:50:16 - [info] 

------------

It seems to happen just as I fix up the username and password to the MQTT Broker. 

Did anyone else have issues after upgrading? 

Thanks,
Alex

--- a/src/nodes/devices-scenic.js
+++ b/src/nodes/devices-scenic.js
@@ -12,6 +12,11 @@ module.exports = function (RED) {
         const regId = bavaria.observer.register(bridgeNode.id + "_connected", function (message) {
             node.status({ fill: "green", text: "connected" });
             bridgeNode.subscribeDevice(node.id, config.deviceName, function (message) {
+                if (message.action === undefined || message.action === "") {
+                    // Ignore message with empty action
+                    return;
+                }
+
                 var ioMap = {
                     recall_scene_0: utils.createButtonOutput(0, "A0", "pressed"),
                     recall_scene_4: utils.createButtonOutput(0, "A0", "released"),

--- a/src/nodes/devices-sonoff.js
+++ b/src/nodes/devices-sonoff.js
@@ -12,7 +12,8 @@ module.exports = function (RED) {
             node.status({ fill: "green", text: "connected" });
             bridgeNode.subscribeDevice(node.id, config.deviceName, function (message) {
                 try {
-                    if (message.action === undefined) {
+                    if (message.action === undefined || message.action === "") {
+                        // Ignore message with empty action
                         return;
                     }
 

--- a/src/nodes/hue-dimmer/devices-hue.js
+++ b/src/nodes/hue-dimmer/devices-hue.js
@@ -11,9 +11,11 @@ module.exports = function (RED) {
         const regId = bavaria.observer.register(bridgeNode.id + "_connected", function (message) {
             node.status({ fill: "green", text: "connected" });
             bridgeNode.subscribeDevice(node.id, config.deviceName, function (message) {
-                if (typeof message.action === 'undefined') {
+                if (message.action === undefined || message.action === "" ||  message.action === 'undefined') {
+                    // Ignore message with empty action
                     return;
                 }
+                
                 message.action = message.action.split("-").join("_");
 
                 const ioMap = {};

--- a/src/nodes/hue-dimmer/devices-hue.js
+++ b/src/nodes/hue-dimmer/devices-hue.js
@@ -11,7 +11,7 @@ module.exports = function (RED) {
         const regId = bavaria.observer.register(bridgeNode.id + "_connected", function (message) {
             node.status({ fill: "green", text: "connected" });
             bridgeNode.subscribeDevice(node.id, config.deviceName, function (message) {
-                if (message.action === undefined || message.action === "" ||  message.action === 'undefined') {
+                if (message.action === undefined || message.action === "") {
                     // Ignore message with empty action
                     return;
                 }

--- a/upcoming-changelog.md
+++ b/upcoming-changelog.md
@@ -1,15 +1,17 @@
-# Changelog for the upcomming version
-> This document is an internal note for the changelog of the upcomming release.
+# Changelog for the upcoming version
+> This document is an internal note for the changelog of the upcoming release.
 
 > If new features are added, increase the minor version || if bug fixes are added, increase the patch version.
 
 ### Release: `0.19.3`
 
 #### Features:
-- Documented additional settings for upgrading to mosquitto 2.0 in the getting started guide.
+- Documented additional settings for upgrading to mosquito 2.0 in the getting started guide.
 - Documented Mirek scale. 
 
 #### Bug fixes:
 - Not set property on Ikea Remote device caused a complete Node-Red restart
+- Not set property on Hue Remote device caused a complete Node-Red restart
+- Preventive measure: Check action for empty string in scenic remote and sonnoff buttons
 
 #### Behind the scenes

--- a/upcoming-changelog.md
+++ b/upcoming-changelog.md
@@ -6,7 +6,7 @@
 ### Release: `0.19.3`
 
 #### Features:
-- Documented additional settings for upgrading to mosquito 2.0 in the getting started guide.
+- Documented additional settings for upgrading to mosquitto 2.0 in the getting started guide.
 - Documented Mirek scale. 
 
 #### Bug fixes:

